### PR TITLE
WIP: Migrate to 0.6+ and HTTP.jl

### PR DESCRIPTION
**Pull request by ararslan**
cc @kshyatt, @quinnj

### CODE REVIEW

**Comment by kshyatt** [comment]
So I'm not at all sure about the VERSION numbers here, I picked the latest version of `master` I'm on and my laptop is too weaksauce to do a proper bisect. Anyone want to take a run at it?

The base parser has changed behaviour a little bit as has the lcov and meminfo dump.

File: src/codecovio.jl [hunk 1]
```diff
@@ -47,13 +47,19 @@ module Codecov
     not already specified in args_array.
     """
     function set_defaults(args_array; kwargs...)
-        defined_names = [k for (k,v) in args_array]
+        defined_names = keys(pairs(args_array))
+        is_args_array = Pair{Symbol, Any}[]
+        if args_array isa Base.Iterators.IndexValue
```
Comment (ararslan) [inline src/codecovio.jl]: `IndexValue` isn't a thing on 0.6.

Comment (kshyatt) [reply src/codecovio.jl]: yeah we should maybe have 2 versions of this function, a 0.7 version and a 0.6 version

File: test/runtests.jl [hunk 2]
```diff
@@ -55,9 +57,9 @@ cd(dirname(@__DIR__)) do
     # another VERSION check.
 
     if (VERSION.major == 0 && VERSION.minor == 7 && VERSION < v"0.7.0-DEV.468") || VERSION < v"0.6.1-pre.93"
-        target = Union{Int64,Void}[nothing, 1, nothing, 0, nothing, 0, nothing, nothing, nothing, 0, nothing, nothing, nothing, nothing, nothing, nothing, 0, nothing, nothing, 0, nothing, nothing, nothing, nothing]
+        target = Union{Int64,Nothing}[nothing, 1, nothing, 0, nothing, 0, nothing, nothing, nothing, 0, nothing, nothing, nothing, nothing, nothing, nothing, 0, nothing, nothing, 0, nothing, nothing, nothing, nothing]
     else
-        target = Union{Int64,Void}[nothing, 1, nothing, 0, nothing, 0, nothing, nothing, nothing, nothing, 0, nothing, nothing, nothing, nothing, nothing, 0, nothing, nothing, 0, nothing, nothing, nothing, nothing]
+        target = Union{Int64,Nothing}[nothing, 1, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing]
```
Comment (ararslan) [inline test/runtests.jl]: This change seems to be incorrect on 0.6.

File: test/runtests.jl [hunk 3]
```diff
@@ -89,10 +89,10 @@ cd(dirname(@__DIR__)) do
     rm(script)
 
     # Test command-line usage
-    if VERSION < v"0.7.0-DEV.3481"
+    if VERSION < v"0.7.0-DEV.3001"
         @test readchomp(`$(Base.julia_cmd()) $(joinpath("src", "Coverage.jl"))`) == "Coverage.MallocInfo[]"
     else
-        @test readchomp(`$(Base.julia_cmd()) $(joinpath("src", "Coverage.jl"))`) == "Main.Coverage.MallocInfo[]"
+        @test readchomp(`$(Base.julia_cmd()) $(joinpath("src", "Coverage.jl"))`) == "Coverage.MallocInfo[]\nMain.Coverage.MallocInfo[]"
```
Comment (ararslan) [inline test/runtests.jl]: Do you know what changed in Base that made it print this way? It seems wrong...

Comment (kshyatt) [reply test/runtests.jl]: No, I don't...

File: src/coveralls.jl [hunk 4]
```diff
@@ -135,9 +135,9 @@ module Coveralls
             end
         end
 
-        r = post(URI("https://coveralls.io/api/v1/jobs"), files =
+        r = HTTP.post("https://coveralls.io/api/v1/jobs", files =
             [FileParam(JSON.json(data),"application/json","json_file","coverage.json")])
         println("Result of submission:")
-        println(Compat.UTF8String(r.data))
+        println(String(r.data))
```
Comment (mortenpi) [inline src/coveralls.jl]: [`HTTP.Response`](https://juliaweb.github.io/HTTP.jl/latest/index.html#HTTP.Messages.Response) apparently does not have a `.data` field. This crashes Codecov (after submission), e.g. [here](https://travis-ci.org/JuliaDocs/Documenter.jl/jobs/332152832#L1234-L1242). Should be `.body` instead?

Comment (ararslan) [reply src/coveralls.jl]: Thanks, good catch. See #147.

**Pull request merged**